Introduce template_dir config option

mintelm opened the pull request:
This introduces a new config option called `template_dir`. What this new config option does it adds the possibility to define an additional directory to include when using `load_template()`. I have the 'issue' that my neovim configuration setup is as follows: all my plugins are under `$RTP/lua/plugins/...`. This means that I can not have my templates right where I want them to be (i.e. `$RTP/lua/plugins/overseer/template/abc`. This patch would fix my issue and I can imagine that others could also benefit from specifying an additional template directory.

stevearc reviewed (changes requested):
What do you think about changing it to `template_dirs` as a list of strings and using it for the definition of the default path?

mintelm commented:
I think that's a nicer approach yes, I changed it and also rebased to latest main.  
I was unsure which format to use for the directories, i.e. `foo/bar` or `foo.bar`, since I am not typically developing lua. What do you prefer? I used the dot style right now (I think this is typically used for loading lua modules via `require()`).
